### PR TITLE
add ReadArchive command

### PR DESCRIPTION
**Pull request by DeadByElpy**

### CODE REVIEW

File: src/obis-observer/commands/downlink/ReadArchive.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+/**
+ * Downlink command.
+ *
+ * @example
+ * ```js
+ * import ReadArchive from 'jooby-codec/obis-observer/commands/downlink/ReadArchive.js';
+ *
+ * const parameters = {
+ *     requestId: 34,
+ *     archiveType: 2,
```
Comment (DarkPark) [inline src/obis-observer/commands/downlink/ReadArchive.ts:50]: would be nice to use constants here as well

File: src/obis-observer/commands/downlink/ReadArchive.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    // returns full message - header with body
+    toBytes (): Uint8Array {
+        if ( typeof this.size !== 'number' ) {
+            throw new Error('unknown or invalid size');
+        }
+
+        const {requestId, archiveType, time} = this.parameters;
+
```
Comment (DarkPark) [inline src/obis-observer/commands/downlink/ReadArchive.ts:95]:
```suggestion
```

**Review by DarkPark — CHANGES_REQUESTED** [review]

File: src/obis-observer/commands/downlink/ReadArchive.ts (hunk 1 above)
Comment (DeadByElpy) [reply src/obis-observer/commands/downlink/ReadArchive.ts:50]: constants added

File: src/obis-observer/commands/downlink/ReadArchive.ts (hunk 2 above)
Comment (DeadByElpy) [reply src/obis-observer/commands/downlink/ReadArchive.ts:95]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**